cernopendata_client/searcher.py

```diff
@@ -96,7 +103,7 @@ def get_recid(server=None, title=None, doi=None):
     response_json = response.json()
     try:
         response.raise_for_status()
-    except requests.HTTPError as e:
+    except Exception as e:
```

Comment:
It would be nice to list all the exceptions instead of one single catch all.

BTW the scenario described in #98 does not for me fully with this PR.  I get:

```
ConnectionResetError: [Errno 104] Connection reset by peer
...
During handling of the above exception, another exception occurred:
...
urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
...
During handling of the above exception, another exception occurred:
...
requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
```

Is this improved with sequel PRs? Shall we address them in one go?

Reply:
Yes, This PR addresses #98 and will be completed in the sequel of PRs as we discussed.

Reply:
So let's perhaps squash them together, it'll be easier to test that way!

Reply:
The general `except` catch-all is not the best paradigm, as it can mask some error situations, but it is OK for now for partial addressing...